pystac_client/item_search.py

```diff
@@ -409,9 +417,15 @@ def _format_fields(self, value: Optional[FieldsLike]) -> Optional[Fields]:
     def _format_intersects(value: Optional[IntersectsLike]) -> Optional[Intersects]:
         if value is None:
             return None
+        if isinstance(value, dict):
+            return deepcopy(value)
         if isinstance(value, str):
             return json.loads(value)
-        return deepcopy(getattr(value, '__geo_interface__', value))
+        if hasattr(value, '__geo_interface__'):
```

Comment:
Can you add a comment to add [runtime_checkable](https://docs.python.org/3/library/typing.html#typing.runtime_checkable) when we drop 3.7 support?

Reply:
I didn't know about that before, thanks